server/db/dbcontrollers/stocksController.js

```diff
@@ -25,5 +25,53 @@ module.exports = function (knex) {
       .orWhere(knex.raw('UPPER(name) like UPPER(?)', [searchLike]));
   };
 
+  //pass in array of objects contaning symbol,shares,and price 
+  //query stock_prices and update price if needed and return
+  module.getPrices = function (stockArr) {
+    
+    return knex.select('symbol','ask')
+      .from('stock_prices')
+    .then(function (dataArr) {
+
+      return stockArr.map(function (stockObj) {
+        var price;
+        for (var i = 0; i < dataArr.length; i++) {
+          if (dataArr[i].symbol === stockObj.stockSymbol) {
+            if (dataArr[i].ask !== stockObj.price) {
+              stockObj.price = dataArr[i].ask;
+            }
+          }
+        }
+        return stockObj;
+      });
+
+    });
+
+  };
+
+  //pass in array of objects contaning symbol,shares,and price 
+  //query stock_prices and update price if needed and return
+  module.updatePrices = function (stockArr) {
+    return knex.select('symbol','ask')
+      .from('stock_prices')
+    .then(function (dataArr) {
+      return stockArr.map(function (stockObj) {
```

Comment:
same comment as above here. If we could use the provided symbols in a where clause, it seems like we could reduce the data we need to get back from the DB dramatically.